src/main/java/teammates/sqllogic/core/AccountRequestsLogic.java

```diff
@@ -0,0 +1,91 @@
+package teammates.sqllogic.core;
+
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidOperationException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.sqlapi.AccountRequestsDb;
+import teammates.storage.sqlentity.AccountRequest;
+
+/**
+ * Handles operations related to account requests.
+ *
+ * @see AccountRequest
+ * @see AccountRequestsDb
+ */
+public final class AccountRequestsLogic {
+
+    private static final AccountRequestsLogic instance = new AccountRequestsLogic();
+
+    private AccountRequestsDb accountRequestDb;
+
+    private AccountRequestsLogic() {
+        // prevent notification
+    }
+
+    public static AccountRequestsLogic inst() {
+        return instance;
+    }
+
+    /**
+     * Initialise dependencies for {@code AccountRequestLogic} object.
+     */
+    public void initLogicDependencies(AccountRequestsDb accountRequestDb) {
+        this.accountRequestDb = accountRequestDb;
+    }
+
+    /**
+     * Creates an account request.
+     */
+    public AccountRequest createAccountRequest(String name, String email, String institute)
+            throws InvalidParametersException, EntityAlreadyExistsException {
+        AccountRequest toCreate = new AccountRequest(email, name, institute);
+
+        return accountRequestDb.createAccountRequest(toCreate);
+    }
+
+    /**
+     * Gets account request associated with the {@code }.
+     */
+    public AccountRequest getAccountRequest(String email, String institute) {
+
+        return accountRequestDb.getAccountRequest(email, institute);
+    }
+
+    /**
+     * Creates/resets the account request with the given email and institute such that it is not registered.
+     */
+    public AccountRequest resetAccountRequest(String email, String institute)
+            throws EntityDoesNotExistException, InvalidOperationException, InvalidParametersException {
+        AccountRequest accountRequest = accountRequestDb.getAccountRequest(email, institute);
+
+        if (accountRequest == null) {
+            throw new EntityDoesNotExistException("Account request for instructor with email: " + email
+            + " and institute: " + institute + " does not exist.");
+        }
+
+        if (accountRequest.getRegisteredAt() == null) {
+            throw new InvalidOperationException("Unable to reset account request as instructor is still unregistered.");
+        }
+
+        accountRequest.setRegisteredAt(null);
+
+        return accountRequestDb.updateAccountRequest(accountRequest);
+    }
+
+    /**
+     * Deletes account request associated with the {@code email} and {@code institute}.
+     *
+     * <p>Fails silently if the account request doesn't exist.</p>
```

Comment:
this does not fail 'silently' since we are throwing an exception, so lets remove this comment

Reply:
Valid comment for this and good spot (we have to change this for other entities too). However, would prefer keeping the original behaviour here (failing silently) for backwards compatibility reasons. This will could break the existing APIs that expect no error to be thrown. Perhaps change the code to fail silently instead?

Reply:
this exception was previously thrown in the action class, so we moved this here to reduce the complexity of the action class (the account is fetch to check, if null -> the action throws exception, else carry out the deletion). But yes this method is used in different places other than the action, so @kevin9foong can you check if this is used anywhere else and update accordingly?

We can keep the original behavior and it's easier to implement that way, but this is a simple change so I think we can update the current code to accommodate this, instead of sticking with the original logic

Reply:
yes, so we need to edit the javadocs. At the same time, this function is throwing an exception now which is not the original behaviour, so we need to check if this is used anywhere else

Reply:
Sure, let me do the necessary checks and update the javadocs.

Reply:
Resolved by improving the explanation of the comment for the scenarios exceptions are thrown.